src/nvim/eval.c

```diff
@@ -12195,18 +12194,23 @@ static int list2proftime(typval_T *arg, proftime_T *tm) FUNC_ATTR_NONNULL_ALL
   }
 
   int error = false;
-  varnumber_T n1 = list_find_nr(arg->vval.v_list, 0L, &error);
-  varnumber_T n2 = list_find_nr(arg->vval.v_list, 1L, &error);
+  int32_t high = list_find_nr(arg->vval.v_list, 0L, &error);
+  int32_t low = list_find_nr(arg->vval.v_list, 1L, &error);
   if (error) {
     return FAIL;
   }
 
+  // When we converted the proftime_T to two 32-bit ints, we shifted the high
+  // part by one and made the sign bit of the low part 0. Undo these changes.
+  low = (high & 1 ? low + INT32_MAX + 1 : low);
```

Comment:
Do you mean that `low` will become negative on `high & 1`? That's intentional so that when we pack it into `*tm`, it will be the same value as returned by `profile_start()`. If you meant something else, please explain.

I assume that `INT32_MAX = 0x7ffff...` so adding one makes it `0x80000...`. Though, I've been thinking I'll change this to 
```c
low |= high & 1 ? INT32_MAX + 1 : 0;
```
to make the relationship more clear.